src/util/cout_message.cpp

```diff
@@ -35,6 +37,35 @@ cerr_message_handlert::cerr_message_handlert():
 {
 }
 
+console_message_handlert::console_message_handlert(bool _always_flush)
+  : always_flush(_always_flush), is_a_tty(false), use_SGR(false)
+{
+#ifdef _WIN32
+  HANDLE out_handle=GetStdHandle(STD_OUTPUT_HANDLE);
+
+  DWORD consoleMode;
+  if(GetConsoleMode(out_handle, &consoleMode))
+  {
+    is_a_tty = true;
+
+    consoleMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+    if(SetConsoleMode(out_handle, consoleMode))
+      use_SGR = true;
+  }
+#else
+  is_a_tty = isatty(STDOUT_FILENO);
+  use_SGR = is_a_tty;
+#endif
+}
+
+std::string console_message_handlert::command(unsigned c) const
```

Comment:
Missing documentation. The name does not in an obvious way relate to it's functionality, and what values `c` should take is a mystery.